### PR TITLE
Trus 2486 - add other individual

### PR DESCRIPTION
**Pull request by JoshuaH01**

### CODE REVIEW

File: app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala [hunk 1]
```diff
@@ -21,49 +21,70 @@ import org.slf4j.LoggerFactory
...
+              utr,
+              index,
+              request.identifier,
+              otherIndividual
+            ) map { _ =>
+              Ok
+            }
+          case JsError(errors) =>
+            logger.warn(s"[OtherIndividualTransformationController][addOtherIndividualTransformer] " +
+              s"Supplied json could not be read as an Unidentified Beneficiary - $errors")
```
Comment (jassalrichy) [inline app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala:64]: addOtherIndividualTransformer ... should be amendOtherIndividualTransformer

**Comment by adamconder** [comment]
@JoshuaH01 there are no changes for DeltaTransformSpec.scala

File: app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala [hunk 2]
```diff
@@ -21,49 +21,70 @@ import org.slf4j.LoggerFactory
...
+            transformService.addOtherIndividualTransformer(
+              utr,
+              request.identifier,
+              otherIndividual
+            ) map { _ =>
+              Ok
+            }
+          case JsError(errors) =>
+            logger.warn(s"[OtherIndividualTransformationController][addOtherIndividualTransformer] " +
+              s"Supplied json could not be read as an Unidentified Beneficiary - $errors")
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala:84]: This log should not be unidentified beneficiary

File: app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala (hunk 1 above)
Comment (adamconder) [reply app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala:64]: The log should also not say unidentified beneficiary

File: it/uk/gov/hmrc/transformations/AddOtherIndividualSpec.scala [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          |                "lastName":"O'Connor"
+          |              },
+          |              "dateOfBirth":"2002-01-01",
+          |              "identification":{
+          |                "nino":"KC456736"
+          |              },
+          |              "entityStart":"1998-02-12"
+          |            }""".stripMargin)
+
+      lazy val expectedGetAfterAddBeneficiaryJson: JsValue =
```
Comment (adamconder) [inline it/uk/gov/hmrc/transformations/AddOtherIndividualSpec.scala:63]: Naming is after add beneficiary?

File: test/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationControllerSpec.scala [hunk 4]
```diff
@@ -135,6 +136,47 @@ class OtherIndividualTransformationControllerSpec extends FreeSpec
...
+        .withHeaders(CONTENT_TYPE -> "application/json")
+
+      val result = controller.addOtherIndividual("aUTR").apply(request)
 
+      status(result) mustBe OK
+      verify(otherIndividualTransformationService).addOtherIndividualTransformer("aUTR", "id", newOtherIndividual)
+    }
+
+    "must return an error for malformed json" in {
+      val beneficiaryTransformationService = mock[BeneficiaryTransformationService]
```
Comment (adamconder) [inline test/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationControllerSpec.scala:170]: This should not be the beneficiary transformation service

File: test/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationControllerSpec.scala [hunk 5]
```diff
@@ -135,6 +136,47 @@ class OtherIndividualTransformationControllerSpec extends FreeSpec
...
+      val controller = new BeneficiaryTransformationController(identifierAction, beneficiaryTransformationService)
```
Comment (adamconder) [inline test/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationControllerSpec.scala:171]: Should not be the beneficiary controller

File: test/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationControllerSpec.scala [hunk 6]
```diff
@@ -135,6 +136,47 @@ class OtherIndividualTransformationControllerSpec extends FreeSpec
...
+
+      val request = FakeRequest("POST", "path")
+        .withBody(Json.parse("{}"))
+        .withHeaders(CONTENT_TYPE -> "application/json")
+
+      val result = controller.addUnidentifiedBeneficiary("aUTR").apply(request)
```
Comment (adamconder) [inline test/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationControllerSpec.scala:177]: Shouldn't be calling the unidentified beneficiary method

File: app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala (hunk 1 above)
Comment (JoshuaH01) [reply app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala:64]: Complete

File: app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala (hunk 2 above)
Comment (JoshuaH01) [reply app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala:84]: see above

File: it/uk/gov/hmrc/transformations/AddOtherIndividualSpec.scala (hunk 3 above)
Comment (JoshuaH01) [reply it/uk/gov/hmrc/transformations/AddOtherIndividualSpec.scala:63]: Done

**Pull request merged**